server/cp/set_byline_on_publish.py

```diff
@@ -1,21 +1,66 @@
 from superdesk.signals import item_publish
+from settings import DEFAULT_AUTHOR_EN, DEFAULT_AUTHOR_FR
+import superdesk
 
 
 def set_byline_on_publish(sender, item, updates, **kwargs):
     updated = item.copy()
     updated.update(updates)
 
-    if updated.get("byline") or not updated.get("authors"):
+    if updated.get("byline"):
         return
 
-    byline = ", ".join([get_author_name(author) for author in item.get("authors", [])])
+    if not updated.get("authors"):
+        language = updated.get("language", "en-CA")
+        default_author_username = (
+            DEFAULT_AUTHOR_EN if language == "en-CA" else DEFAULT_AUTHOR_FR
+        )
 
+        users_service = superdesk.get_resource_service("users")
+        default_user = users_service.find_one(
+            req=None, username=default_author_username
+        )
+
+        if not default_user:
+            raise ValueError(
+                f"Default user '{default_author_username}' not found in the database."
+            )
+
+        byline = default_user.get("username", "Unknown User")
+        default_email = default_user.get("email", "contact@cp.ca")
+
+        item["byline"] = updates["byline"] = byline
+        item["authors"] = updates["authors"] = [
```

Comment:
we should match the usual structure there so the formatters can pick it up later